src/Microsoft.AspNetCore.Authentication.Facebook/FacebookOptions.cs

```diff
@@ -57,6 +60,6 @@ public string AppSecret
         /// The list of fields to retrieve from the UserInformationEndpoint.
         /// https://developers.facebook.com/docs/graph-api/reference/user
         /// </summary>
-        public IList<string> Fields { get; }
+        public ICollection<string> Fields { get; } = new HashSet<string>();
```

Comment:
> The only difference with IList was that it guaranteed ordering, though I don't know if that will matter to any of the providers.

[The order doesn't matter](https://tools.ietf.org/html/rfc6749#section-3.3) and in practice, I've never seen a provider breaking the spec here (well, except for the separator... :smile:):

> The value of the scope parameter is expressed as a list of space-delimited, case-sensitive strings. The strings are defined by the authorization server.  If the value contains multiple space-delimited strings, **their order does not matter**, and each string adds an additional access range to the requested scope.